### ISSUE TITLE
Upgrade to Vue.js 3

### ISSUE THREAD

**Issue by kleinfreund** [kumahq/kuma-gui#359]
### Description

**TL;DR**: Upgrade the project to use Vue.js 3.

---

All dependencies in the project have been updated and we can finally upgrade to Vue.js 3.

### vue-meta

The [vue-meta](https://github.com/nuxt/vue-meta) package is not very actively maintained and not yet compatible with Vue 3. However, it is easy to replace because we don’t use a lot of its features.

The short version: In `App.vue`, we can add a watcher like this:

```js
'$route.meta': function (routeMeta) {
  const siteTitle = `${process.env.VUE_APP_NAMESPACE} Manager`

  document.title = routeMeta?.title ? `${routeMeta.title} | ${siteTitle}` : siteTitle
}
```

### vue-prismjs

The [vue-prismjs](https://github.com/vue-formation/vue-prismjs) package is not actively maintained and also not compatible with Vue 3.

This is more complicated to solve since it involves syntax highlighting of code. Specifically, we have a couple of places in the GUI where JSON/YAML code needs to be rendered and highlighted. Here, we also might want to implement our own component since this would give us the flexibility of adding features to it (recently, a grep-like line filtering was brought up).

## Tests

The largest amount of changes will be needed in our unit test suite. That is because we now need to pull more weight when it comes to mocking a component’s dependencies: stores and routes. While this causes a lot of work, it’s good that we need to do this because it will our tests behave more like actual unit tests. Currently, we rely a lot of global application state and mechanisms. We also commonly render the components fully, including all their child components (regardless of source). This gives external dependencies like kongponents a big influence over the stability of the tests. Adding a single attribute to `KButton`, for example, would break all snapshot tests where `KButton` is used (which is really a lot of them).

**Comment by kleinfreund** [comment]
Implemented in #361.
